src/main/java/org/sefglobal/scholarx/controller/MenteeController.java

```diff
@@ -24,12 +25,11 @@ public MenteeController(MenteeService menteeService) {
         this.menteeService = menteeService;
     }
 
-
     @PutMapping("/{id}/state")
     @ResponseStatus(HttpStatus.OK)
     public Mentee approveOrRejectMentee(@PathVariable long id,
-                                        @Valid @RequestBody boolean isApproved)
+                                        @Valid @RequestBody Map<String, Boolean> payload)
             throws ResourceNotFoundException, BadRequestException {
-        return menteeService.approveOrRejectMentee(id, isApproved);
+        return menteeService.approveOrRejectMentee(id, payload.get("isApproved"));
```

Comment:
Null check before sending. If `isApproved ` is not in the payload a type mismatch error will occure

Reply:
I added the null check in the controller itself. Or is it better if the whole payload is sent to the service method and the check is done in there ?

Reply:
no it's ok to validate the payload in the controller